### ISSUE TITLE
Training code

### ISSUE THREAD

**Issue by natstand** [wyf0912/R2LCM#1]
Your work is very compelling, and how to get the training code?

**Comment by wyf0912** [comment]
Hi, I just upload the training and testing code.

Please feel free to drop a message here if you meet any problem

**Comment by natstand** [comment]
`
class MaskedDeconv(nn.Module):
    def __init__(self,*args,**kwargs):
        super().__init__()
        kernel_size = kwargs.pop("kernel_size")
        padding = kernel_size//2
        self.kernel_size=kernel_size
        self.deconv = nn.ConvTranspose2d(kernel_size=kernel_size, padding=padding, stride=1, *args, **kwargs)
        # self.count = nn.Conv2d(kernel_size=kernel_size, padding=padding, in_channels=1, out_channels=1,stride=1)
        
    def forward(self, input):
        input, mask = input[:,:-1], input[:,[-1]]
        deconved = self.deconv(input)
        assert deconved.shape[2:]==input.shape[2:]
        with torch.no_grad():
            count_mask = F.conv2d(mask, weight=torch.ones(1,1,self.kernel_size, self.kernel_size).to(input.device), padding=self.kernel_size//2)
        deconved = deconved/count_mask.clip(min=1)

        weight_1x1 = self.deconv.weight[:,:,self.kernel_size//2,self.kernel_size//2].unsqueeze(2).unsqueeze(3)
        bias = self.deconv.bias
        resid = F.conv_transpose2d(input, weight_1x1, bias, padding=0)
        deconved = deconved*(1-mask) + resid*mask
        
        return torch.cat([deconved, mask], dim=1)  `

The shape of the input here is 16 channels, and the initial ConvTranspose2d is 19 channels, error

**Comment by wyf0912** [comment]
Hi, sorry for replying late. I just fixed the bug in ```compressai/models/learned_raw_journal4.py```.

You could see the latest commit just now.
